Investigate why CI keeps failing with error "Unable to create debug keystore"

Most of our GitHub Actions have started failing recently with:

> FAILURE: Build failed with an exception.
> 
> What went wrong:
> Execution failed for task ':admob:app:validateSigningDebug'.
> java.io.IOException: Unable to create debug keystore in /home/runner/.config/.android because it is not writable.
> 138 actionable tasks: 86 executed, 13 from cache, 39 up-to-date
> 
> Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
> 
> BUILD FAILED in 3m 46s
> Error: Process completed with exit code 1.

Not entirely sure why, but I'm guessing something changed in GitHub Actions recently.

--- a/.github/workflows/android.yml
+++ b/.github/workflows/android.yml
@@ -6,9 +6,7 @@ on:
 
 jobs:
   build:
-    # TODO: Unpin version once firebase/snippets-android#602 is fixed
-    # https://github.com/firebase/snippets-android/issues/602
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - name: Check Snippets

